scripts/performance/tracer.py

```diff
@@ -81,13 +101,20 @@ def start_as_current_span(self, *top_args, **top_kwargs):
         Returns:
             The result of executing the decorated function.
         """
-        def decorator(func):
-            def wrapper(*args, **kwargs):
-                if self.tracer is not None:
-                    with self.tracer.start_as_current_span(*top_args, **top_kwargs):
-                        return func(*args, **kwargs)
-                else:
+        def decorator(func: Callable[P, R]) -> Callable[P, R]:
+            # If no tracer, just return the original function unchanged (best for typing & introspection).
+            if self._tracer is None:
+                return func
+
+            @functools.wraps(func)
+            def wrapped(*args: P.args, **kwargs: P.kwargs) -> R:
+                if self._tracer is None:
```

Comment:
Do we need both self._tracer is None checks? If not, I think we should remove one of them (probably the one in wrapped).